src/index.js

```diff
@@ -11,14 +11,36 @@
  */
 import { load as loadquerystring } from './loaders/url.js';
 import { load as loadtext } from './loaders/text.js';
+import { load as loadjson } from './loaders/json.js';
 import { adapt as createfilter } from './adapters/filter.js';
 
+/**
+ * Loads a Query Builder AST from generic input: either URL query string (with &)
+ * or as key-value-pairs (with line breaks) or as Object (from JSON or YAML notation)
+ * @param {object|string|string[]} input QBL input
+ * @returns {object} Query Builder AST
+ */
+function load(input) {
+  if ((typeof input === 'object' && !Array.isArray(input)) || (Array.isArray(input) && input.every((e) => typeof e === 'object'))) {
+    return loadjson(input);
+  }
+  if (typeof input === 'string' || (Array.isArray(input) && input.every((e) => typeof e === 'string'))) {
+    const expr = Array.isArray(input) ? input.join('') : input;
+    return expr.indexOf('\n') > 0 ? loadtext(expr) : loadquerystring(expr);
+  }
+  return {};
+}
+
+/**
+ * Convenience function that can be used as a tagged template
+ * function and returns a filter function
+ * @param {string[]} strings string array as used in template literals
+ * @returns {function} filter function that accepts an array and returns a filtered array
+ */
+const filter = (strings) => createfilter(load(strings));
+
 const qb = {
-  filter: (strings) => {
-    const expr = strings.join('');
-    const query = expr.indexOf('\n') > 0 ? loadtext(expr) : loadquerystring(expr);
-    return createfilter(query);
-  },
+  filter,
 };
 
-export { qb };
+export { qb, filter };
```

Comment:
so you export:
- `qb`
- `filter`
- `qb.filter`

why export `qb` at all?

why not just:

```
export default filter;
```

Reply:
I anticipate future exports like `qb.algolia`, `qb.odata`, `qb.sql`, so the default export would break compatibility. Also, default exports don't work well with node 12 (which has ES6 module support), so I try to avoid them.

The question should be: why export `filter` at all? I'm mainly exporting it for the combination of people who prefer

```javascript
const { filter } = require('@adobe/helix-querybuilder');
```

over

```javascript
const { filter: adapt } = require('@adobe/helix-querybuilder/src/adapters/filter.js');
```

but who are using a tree shaker that can't get rid of the other (`qb.algolia`, `qb.odata`, `qb.sql`) properties in `qb`.

Reply:
I don't understand. sorry. 

this is an antipattern and we shouldn't support it:
```
const { filter: adapt } = require('@adobe/helix-querybuilder/src/adapters/filter.js');
```

Reply:
In the world of ES6 modules it is very common to have APIs that are exposed like this:

```
import adapt as filter from '@adobe/helix-querybuilder/src/adapters/filter.js';
```

due to the way how we are distributing `@adobe/helix-querybuilder` this is an option available in any modern browser and enables the selective use of just the loaders and adapters you need, without any build-time or rut-time overhead.

Another group of users will be using `@adobe/helix-querybuilder` in node.js with a bundler. I want to offer them an entry point that is convenient, avoids visceral reactions like "this is an antipattern and we shouldn't support it", is stable, and allows for effective tree shaking.

- only exporting `qb` does not allow for effective tree shaking because all the `qb.filter`, `qb.sql`, etc. functions would be recorded as dependencies and included in the bundle, when only one of the functions would actually be used
- exporting `filter` as the default will break compatibility as soon as we also want to export `sql` or `algolia`